<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>JoystickDialog</class>
 <widget class="QDialog" name="JoystickDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>572</width>
    <height>314</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Joystick</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>-60</x>
     <y>280</y>
     <width>621</width>
     <height>32</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
   </property>
  </widget>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>578</width>
     <height>71</height>
    </rect>
   </property>
   <property name="title">
    <string>Selection and info</string>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_18">
    <item>
     <widget class="QLabel" name="label_16">
      <property name="text">
       <string>Choose Joystick:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QComboBox" name="joystickComboBox"/>
    </item>
    <item>
     <widget class="QPushButton" name="connectPushButton">
      <property name="text">
       <string>&amp;Connect</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="disconnectPushButton">
      <property name="text">
       <string>&amp;Disconnect</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="rescanPushButton">
      <property name="text">
       <string>&amp;Rescan device</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="layoutWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>60</y>
     <width>578</width>
     <height>218</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_19">
    <item>
     <widget class="QGroupBox" name="joystickStateBox">
      <property name="minimumSize">
       <size>
        <width>320</width>
        <height>216</height>
       </size>
      </property>
      <property name="title">
       <string>Joystick States</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_20">
           <item>
            <widget class="QLabel" name="label_17">
             <property name="text">
              <string>X Axis:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_28">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="joystickXaxisLabel">
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="1">
          <spacer name="horizontalSpacer_29">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>18</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="2">
          <layout class="QHBoxLayout" name="horizontalLayout_21">
           <item>
            <widget class="QLabel" name="label_18">
             <property name="text">
              <string>X Rotation:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_30">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="joystickXrotationLabel">
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="1" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_22">
           <item>
            <widget class="QLabel" name="label_19">
             <property name="text">
              <string>Y Axis:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_31">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="joystickYaxisLabel">
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="1" column="1">
          <spacer name="horizontalSpacer_32">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>18</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="2">
          <layout class="QHBoxLayout" name="horizontalLayout_23">
           <item>
            <widget class="QLabel" name="label_20">
             <property name="text">
              <string>Y Rotation:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_33">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="joystickYrotationLabel">
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_24">
           <item>
            <widget class="QLabel" name="label_21">
             <property name="text">
              <string>Z LT Axis:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_34">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="joystickZLTaxisLabel">
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="1">
          <spacer name="horizontalSpacer_35">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>18</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="2">
          <layout class="QHBoxLayout" name="horizontalLayout_25">
           <item>
            <widget class="QLabel" name="label_22">
             <property name="text">
              <string>POV0:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_36">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="joystickPOV0Label">
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="3" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_26">
           <item>
            <widget class="QLabel" name="label_23">
             <property name="text">
              <string>Z RT Axis:</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_37">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="joystickZRTaxisLabel">
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="3" column="1">
          <spacer name="horizontalSpacer_38">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>18</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>13</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_27">
         <item>
          <widget class="QLabel" name="label_24">
           <property name="text">
            <string>Buttons</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_39">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="joystickButtonsTestLabel">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_40">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="joystickInformationBox">
      <property name="minimumSize">
       <size>
        <width>250</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Joystick Information</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_28">
         <item>
          <widget class="QLabel" name="label_25">
           <property name="text">
            <string>Joystick Name:</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_41">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>101</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="joystickNameLabel">
           <property name="text">
            <string>Not connected</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_42">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_29">
         <item>
          <widget class="QLabel" name="label_26">
           <property name="text">
            <string>Joystick Id:                                     </string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_43">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="joystickIdLabel">
           <property name="text">
            <string>-1</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_44">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_30">
         <item>
          <widget class="QLabel" name="label_27">
           <property name="text">
            <string>Number of Buttons:</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_45">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>103</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="joystickButtonsLabel">
           <property name="text">
            <string>-1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_46">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_31">
         <item>
          <widget class="QLabel" name="label_28">
           <property name="text">
            <string>Number of Axis:            </string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_47">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>87</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="joystickAxisLabel">
           <property name="text">
            <string>-1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_48">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_32">
         <item>
          <widget class="QLabel" name="label_29">
           <property name="text">
            <string>Number of Hats:     </string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_49">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>103</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="joystickHatsLabel">
           <property name="text">
            <string>-1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_50">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_33">
         <item>
          <widget class="QLabel" name="label_30">
           <property name="text">
            <string>Number of Balls:    </string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_51">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>103</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="joystickBallsLabel">
           <property name="text">
            <string>-1</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_52">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>21</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>JoystickDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>JoystickDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
